QA/PoroDevTest/src/main/java/common/ui_setup/pom_setup/PoroDevPom/RegistrationPage.java

```diff
@@ -1,89 +1,125 @@
+    public WebElement we_password_errorMsg;
     @FindBy(xpath = "//div[contains(text(),'Please confirm your password')]")
-    public WebElement passwordConfirmation_errorMsg;
+    public WebElement we_passwordConfirmation_errorMsg;
     @FindBy(xpath = "//div[contains(text(),'Please input your department')]")
-    public WebElement department_errorMsg;
+    public WebElement we_department_errorMsg;
     @FindBy(xpath = "//div[contains(text(),'Please input your position')]")
-    public WebElement position_errorMsg;
+    public WebElement we_position_errorMsg;
+
+    //REQUIRED MESSAGE SECTION
+    @FindBy(xpath = "//span[contains(text(),'First name is required')]")
+    WebElement we_firstName_requiredMsg;
+    @FindBy(xpath = "//span[contains(text(),'Last name is required')]")
+    WebElement we_lastName_requiredMsg;
+    @FindBy(xpath = "//span[contains(text(),'Email is required')]")
+    WebElement we_email_requiredMsg;
+    @FindBy(xpath = "//span[contains(text(),'Password is required')]")
+    WebElement we_password_requiredMsg;
+    @FindBy(xpath = "//*[@id=\"registerForm\"]/div[5]/span[2]")
+    WebElement we_confirmPass_requiredMsg;
+    @FindBy(xpath = "//*[@id=\"registerForm\"]/div[5]/span[2]")
+    WebElement we_confirmPass_errorMsg_while_pass_isNotTyped;
+    @FindBy(xpath = "//span[contains(text(),'Department is required')]")
+    WebElement we_department_requiredMsg;
+    @FindBy(xpath = "//span[contains(text(),'Position is required')]")
+    WebElement we_position_requiredMsg;
 
     //SUCCESSFUL MESSAGE SECTION
     @FindBy(xpath = "//span[contains(text(),'Successful registration')]")
     public WebElement successful_registrationMsg;
 
-    public void enterFirstName(String firstName) {
-        elementControl.sendText(firstName_field, firstName);
+    public void registerUser(String firstName, String lastName, String email, String password, String confirmPass, String department, String position) throws InterruptedException {
+        BasePage.clickElement(we_createNewAccount_button);
+        Thread.sleep(2000);
+        BasePage.sendText(we_firstName_inputField, firstName);
+        BasePage.sendText(we_lastName_inputField, lastName);
+        BasePage.sendText(we_email_inputField, email);
+        BasePage.sendText(we_password_inputField, password);
+        BasePage.sendText(we_confirmPass_inputField, confirmPass);
+        BasePage.sendText(we_department_inputField, department);
+        BasePage.sendText(we_position_inputField, position);
+        BasePage.clickElement(we_register_button);
+        Thread.sleep(3000);
+    }
+
+    //ASSERT METHODS
+    public void assert_user_registration_isSuccessful(String expectedResults) {
+        Assert.assertEquals(BasePage.getTextFromElement(successful_registrationMsg), expectedResults);
     }
-    public void enterLastName(String lastName) {
-        elementControl.sendText(lastName_field, lastName);
+
+    public void assert_firstName_isRequired(String expectedResults) {
```

Comment:
No need for this many methods (firstName/lastName/email...). You can loop everything in one method.

Reply:
I will try to implement loop here if I catch all elements in list.

Reply:
Done